src/hazmat/sieve.rs

```diff
@@ -385,4 +383,19 @@ mod tests {
         assert!(format!("{s:?}").starts_with("Sieve"));
         assert_eq!(s.clone(), s);
     }
+
+    #[test]
+    fn sieve_with_max_start() {
+        let start = U64::MAX;
+        let mut sieve = Sieve::new(&start, NonZeroU32::new(U64::BITS).unwrap(), false);
+        assert!(sieve.next().is_none());
+    }
+
+    #[test]
+    fn sieve_with_max_start_and_malicious_manipulation() {
```

Comment:
I don't think people manipulating private attributes is something we need to protect or test against.

Reply:
Yeah that's a fair objection. What I was going for here was a test to guard against changes in the logic that increments the `sieve.incr` member, but perhaps that is overly defensive?

Reply:
I just think it opens a can of worms, because there's a lot of ways we can break things if we're allowed to mutate private attributes to mess with invariants. Could this check be done by a `debug_assert` somewhere?

Reply:
Ok, that's fair. I'll remove this test. :)